File: user-service/src/main/java/rs/edu/raf/repository/CodeRepository.java
package rs.edu.raf.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import rs.edu.raf.model.code.Code;
import rs.edu.raf.model.code.CodeType;

import java.util.Optional;

public interface CodeRepository extends JpaRepository<Code,Long> {
    Optional<Code> findByUserEmailAndCodeType(String email, CodeType codeType);
}

File: user-service-kotlin/src/main/kotlin/org/example/userservicekotlin/repository/CodeRepository.kt
package org.example.userservicekotlin.repository

import org.example.userservicekotlin.model.code.Code
import org.example.userservicekotlin.model.code.CodeType
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface CodeRepository : JpaRepository<Code, Long> {
    fun findByUserEmailAndCodeType(email: String, codeType: CodeType): Optional<Code>
}